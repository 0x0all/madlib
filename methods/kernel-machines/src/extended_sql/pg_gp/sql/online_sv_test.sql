-- Example usage for regression:
delete from sv_results;
delete from sv_model;

select generateRegData(1000, 5);
insert into sv_results (select 'myexp', online_sv_reg_agg(ind, label) from sv_train_data);
select storeModel('myexp');
select svs_predict('myexp', '{1,2,4,20,10}');
   
-- To learn multiple support vector regression models
insert into sv_results (select 'myexp' || gp_segment_id, online_sv_reg_agg(ind, label) from sv_train_data group by gp_segment_id);
select storeModel('myexp', 2); -- n is the number of segments FIX ME
select * from svs_predict_combo('myexp', 2, '{1,2,4,20,10}');

-- Example usage for classification:
select generateClData(2000, 5);
insert into sv_results (select 'myexpc', online_sv_cl_agg(ind, label) from sv_train_data);
select storeModel('myexpc');
select svs_predict('myexpc', '{10,-2,4,20,10}');
   
-- To learn multiple support vector models, replace the above by 
insert into sv_results (select 'myexpc' || gp_segment_id, online_sv_cl_agg(ind, label) from sv_train_data group by gp_segment_id);
select storeModel('myexpc', 2); -- n is the number of segments FIX ME
select * from svs_predict_combo('myexpc', 2, '{10,-2,4,20,10}');

-- Example usage for novelty detection:
select generateNdData(100, 2);
insert into sv_results (select 'myexpnd', online_sv_nd_agg(ind) from sv_train_data);
select storeModel('myexpnd');
select svs_predict('myexpnd', '{10,-10}');  
select svs_predict('myexpnd', '{-1,-1}');  