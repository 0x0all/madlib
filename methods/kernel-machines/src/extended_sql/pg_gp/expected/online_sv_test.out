-- Example usage for regression:
delete from MADLIB_SCHEMA.sv_results;
delete from MADLIB_SCHEMA.sv_model;
select setseed(0.5);
  setseed   
------------
 1073741823
(1 row)

select MADLIB_SCHEMA.generateRegData(1000, 5);
 generateregdata 
-----------------
 
(1 row)

select MADLIB_SCHEMA.sv_regression('MADLIB_SCHEMA.sv_train_data', 'myexp', false);
 sv_regression 
---------------
 
(1 row)

select MADLIB_SCHEMA.svs_predict('myexp', '{1,2,4,20,10}');
   svs_predict    
------------------
 49.8912368092726
(1 row)

select MADLIB_SCHEMA.svs_predict('myexp', '{1,2,4,20,-10}');
    svs_predict    
-------------------
 -51.0059555103496
(1 row)

   
-- To learn multiple support vector regression models
select MADLIB_SCHEMA.sv_regression('MADLIB_SCHEMA.sv_train_data', 'myexp', true);
 sv_regression 
---------------
 
(1 row)

select * from MADLIB_SCHEMA.svs_predict_combo('myexp', '{1,2,4,20,10}');
 model  |    prediction    
--------+------------------
 myexp0 | 58.4043992289992
 myexp1 | 72.4720154277559
 avg    | 65.4382073283776
(3 rows)

-- Example usage for classification:
select MADLIB_SCHEMA.generateClData(2000, 5);
 generatecldata 
----------------
 
(1 row)

select MADLIB_SCHEMA.sv_classification('MADLIB_SCHEMA.sv_train_data', 'myexpc', false);
 sv_classification 
-------------------
 
(1 row)

select MADLIB_SCHEMA.svs_predict('myexpc', '{10,-2,4,20,10}');
   svs_predict    
------------------
 20.9159420459773
(1 row)

   
-- To learn multiple support vector models, replace the above by 
select MADLIB_SCHEMA.sv_classification('MADLIB_SCHEMA.sv_train_data', 'myexpc', true);
 sv_classification 
-------------------
 
(1 row)

select * from MADLIB_SCHEMA.svs_predict_combo('myexpc', '{10,-2,4,20,10}');
  model  |    prediction    
---------+------------------
 myexpc0 |  8.9816044037686
 myexpc1 | 9.13577055449154
 avg     | 9.05868747913007
(3 rows)

-- Example usage for novelty detection:
select MADLIB_SCHEMA.generateNdData(100, 2);
 generatenddata 
----------------
 
(1 row)

select MADLIB_SCHEMA.sv_novelty_detection('MADLIB_SCHEMA.sv_train_data', 'myexpnd', false);
 sv_novelty_detection 
----------------------
 
(1 row)

select MADLIB_SCHEMA.svs_predict('myexpnd', '{10,-10}');  
   svs_predict    
------------------
 -2.6689826301299
(1 row)

select MADLIB_SCHEMA.svs_predict('myexpnd', '{-1,-1}');  
   svs_predict    
------------------
 -2.4142130643595
(1 row)

